addons/account/models/account_move.py

```diff
@@ -2657,6 +2657,14 @@ def _onchange_quick_edit_line_ids(self):
         ):
             return
         self._check_total_amount(self.quick_edit_total_amount)
+    
+    @api.onchange('line_ids')
+    def _onchange_check_taxes(self):
+        previous_tax_lines  = self._origin.line_ids.filtered(lambda line: line.display_type == 'tax')
+        current_tax_lines = self.line_ids._origin.filtered(lambda line: line.display_type == 'tax')
+        if previous_tax_lines != current_tax_lines and self.line_ids.tax_ids:
+            self.line_ids = self._origin.line_ids
```

Comment:
I am unaware of any existing action allowing to delete several lines at the same time in the Journal Entry form view, but, if there is, and if we want to keep the tax lines and go on with the other changes, this line could be replaced by `self.line_ids += (previous_tax_lines - current_tax_lines)`